frontend/simple/views/containers/Proposals.vue

```diff
@@ -55,138 +61,130 @@
-              against: 'Keep 80%'
-            },
-            members: 7,
-            votes: 0,
-            userVote: null
-          }
-        },
-        {
-          type: 'rule',
-          data: {
-            value: 0.92,
-            title: 'Rule: Change Mincome',
-            text: 'Karl had proposed to change <strong>Mincome Rule from 87% to 92%</strong>.',
-            textDetails: 'Instead of 6, now at least 5 of 7 members need to approve a new member.',
-            ctas: {
-              for: 'Change to 92%',
-              against: 'Keep 87%'
-            },
-            members: 7,
-            votes: 1,
-            userVote: false
-          }
-        },
-        {
-          type: 'member',
-          data: {
-            member: {
-              picture: 'assets/images/default-avatar.png',
-              name: 'Kim'
-            },
-            title: 'Add Member',
-            text: 'Rachel had proposed to <strong>add Kim</strong> to the group.',
-            ctas: {
-              for: 'Invite Kim',
-              against: 'Don\'t invite'
-            },
-            members: 7,
-            votes: 3,
-            userVote: true
-          }
-        },
-        {
-          type: 'mincome',
-          data: {
-            value: '$150',
-            title: 'Min Income',
-            text: 'You had proposed to change <strong>mincome from $200 to $150</strong>',
-            members: 7,
-            votes: 0,
-            ownProposal: true
-          }
-        }
-      ]
-    }
+    Voting
   },
   computed: {
-    groupProposals () {
+    ...mapGetters([
+      'currentUserIdentityContract',
+      'currentGroupState'
+    ]),
+    proposalList () {
       const notVoted = []
       const alreadyVoted = []
       const own = []
 
-      for (var i = 0, l = this.mockProposals.length; i < l; i++) {
-        const proposal = this.mockProposals[i]
-        if (proposal.data.ownProposal) {
-          own.push(proposal)
-        } else if (proposal.data.userVote === null) {
-          notVoted.push(proposal)
-        } else if (proposal.data.userVote !== null) {
-          alreadyVoted.push(proposal)
+      const groupData = this.currentGroupState
+      const proposals = groupData.proposals
+      const userData = this.currentUserIdentityContract.attributes
+
+      for (let hash in proposals) {
+        const proposalData = {
+          type: proposals[hash].type, // 'invitation' or 'removal' for member, field name for rule/mincome
+          votes: {
+            total: Object.entries(groupData.profiles).length,
+            received: proposals[hash].for.length + proposals[hash].against.length,
+            threshold: proposals[hash].threshold
+          },
+          value: proposals[hash].value || proposals[hash].candidate || null,
+          originalValue: groupData[proposals[hash].type] || null,
+          ownVote: proposals[hash].for.includes(userData.name) || proposals[hash].against.includes(userData.name)
```

Comment:
I'm not sure... but I remember a while ago when I asked some questions to @taoeffect regarding how voting system worked and one of the questions was

> Should the user know who already voted?

And Greg said no, only the number of votes.

If the user is a developer and debug our code... in this exactly line she/he will know who are the members that already voted. Is this a concern?

Reply:
> If the user is a developer and debug our code... in this exactly line she/he will know who are the members that already voted and what did they voted. Is this a concern?

Nope, not a serious concern :)

Reply:
that information is already stored in the vue store, and in localstorage... we should also change how we cast and store those votes if this is a concern!